src/meltano/core/utils/__init__.py

```diff
@@ -451,5 +451,25 @@ def hash_sha256(value: str) -> str:
 
     Returns:
         The hashed value of the given string.
+
+    Raises:
+        ValueError: If we are blindly passed a value that is None.
     """
+    if value is None:
+        raise ValueError("Cannot hash None.")
     return hashlib.sha256(value.encode()).hexdigest()
+
+
+def safe_hasattr(obj: Any, name: str) -> bool:
+    """Safely checks if an object has a given attribute.
+
+    This is a hacky workaround for the fact that `hasattr` is not allowed by WPS.
+
+    Args:
+        obj: The object to check.
+        name: The name of the attribute to check.
+
+    Returns:
+        True if the object has the attribute, False otherwise.
+    """
+    return getattr(obj, name, None) is not None
```

Comment:
This doesn't work when the attribute being checked has the value `None`. Can we just do what `hasattr` does under the hood instead?

```suggestion
    try:
        getattr(obj, name)
    except AttributeError:
        return False
    return True
```